internal/host/validator.go

```diff
@@ -795,3 +795,21 @@ func (v *validator) printSufficientPacketLossRequirementForRole(c *validationCon
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) isDomainNameResolved(c *validationContext) ValidationStatus {
```

Comment:
Since we're not only checking whether it was resolved, we're checking if it was resolved *correctly*, then maybe consider changing the name to `isDomainNameResolvedCorrectly`